theano/tensor/nnet/abstract_conv.py

```diff
@@ -2311,19 +2421,18 @@ def perform(self, node, inp, out_):
                           for i in range(self.convdim))
 
         if mode == "full":
-            mode = tuple(dil_shape[i] - 1 for i in range(self.convdim))
+            mode = tuple((dil_shape[i] - 1,) * 2 for i in range(self.convdim))
         elif mode == "half":
-            mode = tuple(dil_shape[i] // 2 for i in range(self.convdim))
+            mode = tuple((dil_shape[i] // 2,) * 2 for i in range(self.convdim))
```

Comment:
Idem: I think this can be combined with the previous if. (And maybe extracted to a shared helper function.)